app/services/users/safe_remote_profile_image_url.rb

```diff
@@ -2,9 +2,9 @@ module Users
   module SafeRemoteProfileImageUrl
     # Basic check for nil and blank URLs, alongside likely incomplete URLs, such as just "image.jpg".
     def self.call(url)
-      return url if url.to_s.start_with?("https")
+      url.gsub!("http://", "https://") if url.match?(%r{https?://})
 
-      Users::ProfileImageGenerator.call
+      url.presence || Users::ProfileImageGenerator.call
```

Comment:
the old logic here was
"urls that start with https:// are okay, otherwise call ProfileImageGenerator"

The new logic is 

"urls that start with http:// should start with https://, and any truthy url should be returned if provided, otherwise call the profile image generator" 

Is that correct?

Reply:
Those are good points. I've updated the logic to be more similar to the previous version, since I didn't think through all the edge cases the first time around.